<?xml version="1.0"?>
<launch>

  <arg name="sim" default="true" />

  <!-- Remap force/torque sensor -->
  <remap if="$(arg sim)" from="/gazebo_ft_sensor" to="/ft_sensor"/>
  <remap unless="$(arg sim)" from="/ethdaq_data" to="/ft_sensor"/>

  <!-- Load real ft sensor if needed -->
  <include unless="$(arg sim)" file="$(find artificial_hands_bringup)/launch/onrobot_ft_sensor_bringup.launch">
    <arg name="frame_id" value="robotiq_ft_frame_id"/>
  </include>

  <!-- Load real or simulated (Gazebo) hardware -->
  <include file="$(find artificial_hands_bringup)/launch/mia_hand_on_ur5_bringup.launch" pass_all_args="true"/>
  
  <!-- Load moveit planning and execution interface -->
  <include file="$(find mia_hand_on_ur5_moveit_config)/launch/mia_hand_on_ur5_planning_and_execution.launch" pass_all_args="true"/>

  <!-- Run the handover wrist node -->
  <node pkg="artificial_hands_demos" type="handover_wrist_node" name="handover_wrist_node" output="screen"/>

  <!-- Run the handover state machine -->
  <!-- <node pkg="artificial_hands_demos" type="handover_release_smach.py" name="handover_release_smach" output="screen"/> -->

</launch>